### PR TITLE
Fix invalid prop type for isValidSection warning

### PR DESCRIPTION
**Pull request by Pytal**
Fix invalid prop type for isValidSection warning

```
[Vue warn]: Invalid prop: type check failed for prop "isValidSection". Expected Boolean, got String with value "English (US)".
```

### CODE REVIEW

File: apps/settings/src/utils/validate.js [hunk 1]
```diff
@@ -64,5 +64,5 @@ export function validateEmail(input) {
  */
 export function validateLanguage(input) {
 	return input.code !== ''
-		&& input.name
+		&& Boolean(input.name)
```
Comment (kesselb) [inline apps/settings/src/utils/validate.js:67]: Isn't that `!== ''` just harder to read :thinking:

Comment (oltl4ever81) [reply apps/settings/src/utils/validate.js:67]: -

Comment (Pytal) [reply apps/settings/src/utils/validate.js:67]: The reasoning was that input.code would always be guaranteed to be a string so the check is explicitly for a string, while name uses a property access here https://github.com/nextcloud/server/blob/fix/boolean-expected/apps/settings/src/components/PersonalInfo/LanguageSection/Language.vue#L125 which may be undefined :)

Comment (kesselb) [reply apps/settings/src/utils/validate.js:67]: Thanks and good to know ;)

**Review by julien-nc — COMMENTED** [review]
Works fine.

We could fix the jsdoc for `validateLanguage`, it seems the `input` param is not a String but an Object.

Wouldn't it be more explicit to replace the condition by the following one?
``` js
return input.code !== '' && input.name !== undefined && input.name !== ''
```

**Comment by szaimen** [comment]
Is there a testing procedure? I'd like to test this but don't know how. Can you write some steps?

**Comment by eneiluj** [comment]
> Is there a testing procedure? I'd like to test this but don't know how. Can you write some steps?

@szaimen To witness the warning, with the master branch, you can build the dev target for the settings module. Vue warnings don't show up if you execute the `build` target:

``` bash
cd /var/www/.../nextcloud.dev.git/
git pull origin master
git checkout master
# optionally make sure js packages are up to date
MODULE=settings npm ci
MODULE=settings npm run dev
```
Then load the personal user settings page, you should see the warning in the browser console log.

Then use this PR's branch, build again and reload the settings:

``` bash
git checkout fix/boolean-expected
MODULE=settings npm run dev
```

The warning should be gone and the language settings validation should still work fine.
The list of available JS modules can be found in `/webpack.common.js`.

**Comment by szaimen** [comment]
Thanks for the detailed instructions but the compiling fails for me, unfortunatley. 
I will not investigate this though as it is currently not on my priority list but maybe at some point in the future :)

**Comment by Pytal** [comment]
> We could fix the jsdoc for `validateLanguage`, it seems the `input` param is not a String but an Object.

Nice catch, fixed!

> Wouldn't it be more explicit to replace the condition by the following one?
> 
> ```js
> return input.code !== '' && input.name !== undefined && input.name !== ''
> ```
Updated :)

**Review by julien-nc — APPROVED** [review]
:+1:

**Comment by szaimen** [comment]
I think this branch needs a rebase to get rid of the conflicts...

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**